src/Uno.UI.RuntimeTests/Tests/Windows_Storage/Given_StorageFolder_Native.base.cs

```diff
@@ -0,0 +1,1249 @@
+			var folderName = Guid.NewGuid().ToString();
+			StorageFolder? createdFolder = null;
+			try
+			{
+				createdFolder = await rootFolder.CreateFolderAsync(folderName);
+				Assert.AreEqual(folderName, createdFolder.Name);
+			}
+			finally
+			{
+				if (createdFolder != null)
+				{
+					await createdFolder.DeleteAsync();
+				}
+				await CleanupRootFolderAsync();
+			}
+		}
+
+		[TestMethod]
+		public async Task When_CreateFolder_DisplayName_Matches()
+		{
+			var rootFolder = await GetRootFolderAsync();
+			var folderName = Guid.NewGuid().ToString();
+			StorageFolder? createdFolder = null;
+			try
+			{
+				createdFolder = await rootFolder.CreateFolderAsync(folderName);
+				Assert.AreEqual(folderName, createdFolder.DisplayName);
+			}
+			finally
+			{
+				if (createdFolder != null)
+				{
+					await createdFolder.DeleteAsync();
+				}
+				await CleanupRootFolderAsync();
+			}
+		}
+
+		[TestMethod]
+		public async Task When_CreateFolder_With_File_Like_Name()
+		{
+			var rootFolder = await GetRootFolderAsync();
+			var folderName = GetRandomTextFileName();
+			StorageFolder? createdFolder = null;
+			try
+			{
+				createdFolder = await rootFolder.CreateFolderAsync(folderName);
+				Assert.AreEqual(folderName, createdFolder.Name);
+				Assert.AreEqual(folderName, createdFolder.DisplayName);
+			}
+			finally
+			{
+				if (createdFolder != null)
+				{
+					await createdFolder.DeleteAsync();
+				}
+				await CleanupRootFolderAsync();
+			}
+		}
+
+		[TestMethod]
+		public async Task When_CreateFolder_OfType()
+		{
+			var rootFolder = await GetRootFolderAsync();
+			var folderName = Guid.NewGuid().ToString();
+			StorageFolder? createdFolder = null;
+			try
+			{
+				createdFolder = await rootFolder.CreateFolderAsync(folderName);
+				Assert.IsTrue(createdFolder.IsOfType(StorageItemTypes.Folder));
+			}
+			finally
+			{
+				if (createdFolder != null)
+				{
+					await createdFolder.DeleteAsync();
+				}
+				await CleanupRootFolderAsync();
+			}
+		}
+
+		[TestMethod]
+		public async Task When_CreateFolder_Provider_Matches()
+		{
+			var rootFolder = await GetRootFolderAsync();
+			var folderName = Guid.NewGuid().ToString();
+			StorageFolder? createdFolder = null;
+			try
+			{
+				createdFolder = await rootFolder.CreateFolderAsync(folderName);
+				Assert.AreEqual(rootFolder.Provider, createdFolder.Provider);
+			}
+			finally
+			{
+				if (createdFolder != null)
+				{
+					await createdFolder.DeleteAsync();
+				}
+			}
+		}
+
+		[TestMethod]
+		public async Task When_CreateFolder_Duplicate_Default()
+		{
+			var rootFolder = await GetRootFolderAsync();
+			var folderName = Guid.NewGuid().ToString();
+			StorageFolder? createdFolder = null;
+			try
+			{
+				createdFolder = await rootFolder.CreateFolderAsync(folderName);
+				try
+				{
+					await rootFolder.CreateFolderAsync(folderName);
+					Assert.Fail("Expected exception, no exception thrown.");
+				}
+				catch (Exception)
```

Comment:
Restrict this a bit more?
```cs
catch (Exception e) when (e.Message /*startwith, contains, regex.ismatch...*/)
```

Reply:
The reason I did it this way is that UWP just throws some generic COMException. However, I just realized I can improve it, will try something :-) .